extensions_admin/pulp_docker/extensions/admin/upload.py

```diff
@@ -64,3 +72,69 @@ def generate_override_config(self, **kwargs):
             override_config[constants.CONFIG_KEY_MASK_ID] = kwargs[OPT_MASK_ANCESTOR_ID.keyword]
 
         return override_config
+
+
+class TagUpdateCommand(UploadCommand):
+    """
+    Comamnd used to point a tag to a particular manifest. This will either
+    update an existing tag or create a new tag if one does not exist
+    """
+
+    def __init__(self, context):
+        super(TagUpdateCommand, self).__init__(context, name='tag',
+                                               upload_files=False,
+                                               description=DESC_UPDATE_TAGS)
+        self.add_option(TAG_NAME_OPTION)
+        self.add_option(MANIFEST_DIGEST_OPTION)
+
+    def determine_type_id(self, filename, **kwargs):
+        """
+        Returns the ID of the type of file being uploaded.
+
+        :param filename: full path to the file being uploaded
+        :type  filename: str
+        :param kwargs: arguments passed into the upload call by the user
+        :type  kwargs: dict
+
+        :return: ID of the type of file being uploaded
+        :rtype:  str
+        """
+
+        return constants.TAG_TYPE_ID
+
+    def generate_unit_key(self, filename, **kwargs):
+        """
+        Returns the unit key that should be specified in the upload request.
+
+        :param filename: full path to the file being uploaded
+        :type  filename: str, None
+        :param kwargs: arguments passed into the upload call by the user
+        :type  kwargs: dict
+
+        :return: unit key that should be uploaded for the file
+        :rtype:  dict
+        """
+
+        unit_key = {}
+        metadata = {}
+
+        return unit_key, metadata
```

Comment:
This method should return a unit key consisting of tag name and repo_id. It's defined here https://github.com/pulp/pulp_docker/blob/master/plugins/pulp_docker/plugins/models.py#L280